port/unix/omrmmap.c

```diff
@@ -50,11 +49,97 @@
 #include "protect_helpers.h"
 #include "omrportpriv.h"
 
+#if defined(J9ZOS390)
+#include "omrvmem.h"
+#if !defined(MAP_FAILED)
+#define MAP_FAILED ((void*)-1)
+#endif /* !defined(MAP_FAILED) */
+#endif /* defined(J9ZOS390) */
+
 #if defined(AIXPPC)
 #include <sys/shm.h>
 #include <sys/vminfo.h>
 #endif/*AIXPPC*/
 
+#if defined(J9ZOS390)
+static J9MmapHandle *
+omrmmap_read_map_file(struct OMRPortLibrary *portLibrary, intptr_t file, uint64_t offset, uintptr_t size, uint32_t category)
+{
+	/* default implementation will allocate memory and read the file into it */
+	uintptr_t numBytesRead;
+	intptr_t rc;
+	void *mappedMemory;
+	void *allocPointer;
+	J9MmapHandle *returnVal;
+
+	Trc_PRT_mmap_map_file_default_entered(file, size);
+
+	if (file == -1) {
+		Trc_PRT_mmap_map_file_default_badfile();
+		return NULL;
+	}
+
+	if (0 == size) {
+		struct stat buf;
+
+		memset(&buf, 0, sizeof(struct stat));
```

Comment:
Suggest `memset(&buf, 0, sizeof(buf));` which works whatever type buf has.

Reply:
Changed it to `memset(&buf, 0, sizeof(buf));`